meson_cpu/x86/meson.build

```diff
@@ -0,0 +1,217 @@
+  endforeach
+  SSE42.update(args: {'val': '/arch:SSE4.2', 'match': '/arch:.*'})
+  FMA3.update(args: {'val': '/arch:CORE-AVX2', 'match': '/arch:.*'})
+  AVX2.update(args: {'val': '/arch:CORE-AVX2', 'match': '/arch:.*'})
+  AVX512F.update(args: {'val': '/Qx:COMMON-AVX512', 'match': '/arch:.*'})
+  AVX512CD.update(args: {'val': '/Qx:COMMON-AVX512', 'match': '/arch:.*'})
+  AVX512_KNL.update(args: {'val': '/Qx:KNL', 'match': '/[arch|Qx]:.*'})
+  AVX512_KNM.update(args: {'val': '/Qx:KNM', 'match': '/[arch|Qx]:.*'})
+  AVX512_SKX.update(args: {'val': '/Qx:SKYLAKE-AVX512', 'match': '/[arch|Qx]:.*'})
+  AVX512_CLX.update(args: {'val': '/Qx:CASCADELAKE', 'match': '/[arch|Qx]:.*'})
+  AVX512_CNL.update(args: {'val': '/Qx:CANNONLAKE', 'match': '/[arch|Qx]:.*'})
+  AVX512_ICL.update(args: {'val': '/Qx:ICELAKE-CLIENT', 'match': '/[arch|Qx]:.*'})
+endif
+if compiler_id == 'intel'
+  clear_m = '^(-mcpu=|-march=)'
+  clear_any = '^(-mcpu=|-march=|-x[A-Z0-9\-])'
+  FMA3.update(args: {'val': '-march=core-avx2', 'match': clear_m})
+  AVX2.update(args: {'val': '-march=core-avx2', 'match': clear_m})
+  AVX512F.update(args: {'val': '-march=common-avx512', 'match': clear_m})
+  AVX512CD.update(args: {'val': '-march=common-avx512', 'match': clear_m})
+  AVX512_KNL.update(args: {'val': '-xKNL', 'match': clear_any})
+  AVX512_KNM.update(args: {'val': '-xKNM', 'match': clear_any})
+  AVX512_SKX.update(args: {'val': '-xSKYLAKE-AVX512', 'match': clear_any})
+  AVX512_CLX.update(args: {'val': '-xCASCADELAKE', 'match': clear_any})
+  AVX512_CNL.update(args: {'val': '-xCANNONLAKE', 'match': clear_any})
+  AVX512_ICL.update(args: {'val': '-xICELAKE-CLIENT', 'match': clear_any})
+endif
+if compiler_id == 'msvc'
+  # MSVC Compiler doesn't support the following features
+  foreach fet : [AVX512_KNL, AVX512_KNM]
+    fet.update(disable: compiler_id + ' compiler does not support it')
+  endforeach
+  # The following features don't own private FLAGS still
+  # the compiler provides ISA capability for them.
+  foreach fet : [
+    SSE3, SSSE3, SSE41, POPCNT, SSE42, AVX, F16C, XOP, FMA4,
+    AVX512F, AVX512CD, AVX512_CLX, AVX512_CNL,
+    AVX512_ICL
+  ]
+    fet.update(args: '')
+  endforeach
+  # MSVC compiler doesn't support the following features independently
+  FMA3.update(implies: [F16C, AVX2])
+  AVX2.update(implies: [F16C, FMA3])
+  AVX512F.update(implies: [AVX2, AVX512CD, AVX512_SKX])
+  AVX512CD.update(implies: [AVX512F, AVX512_SKX])
+  clear_arch = '/arch:.*'
+  # only available on 32-bit. Its enabled by default on 64-bit mode
+  foreach fet : [SSE, SSE2]
+    if cpu_family == 'x86'
+      fet.update(args: {'val': '/arch:' + fet.get('name'), 'match': clear_arch})
+    else
+      fet.update(args: '')
+    endif
+  endforeach
+  FMA3.update(args: {'val': '/arch:AVX2', 'match': clear_arch})
+  AVX2.update(args: {'val': '/arch:AVX2', 'match': clear_arch})
+  AVX512_SKX.update(args: {'val': '/arch:AVX512', 'match': clear_arch})
+endif
+
+X86_FEATURES = {
```

Comment:
From AVX-512 onwards, we could simplify this list quite a bit. We pretty much only care about `AV512_SKX, AVX512_ICL and AVX512_SPR`. Rest of them are irrelevant.

Reply:
Agree, I will address it in another pr, also covers the same modifications for distutils.